Stratis.SmartContracts/UInt128.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Numerics;
+
+namespace Stratis.SmartContracts
+{
+    public class UInt128 : UIntBase
```

Comment:
Types used for contracts are struct as design like Address type. It is also BigInteger is struct as well. I think there are good reason that it is intentionally designed as struct.